docs/sources/clients/promtail/logrotation.md

```diff
@@ -0,0 +1,137 @@
+# Promtail and Log Rotation
+
+## Why does log rotation matters?
+
+At any point in time, there may be three processes working on a log file as shown in the image below.
+![block_diagram](./logrotation-components.png)
+
+1. Appender - A writer that keeps appending to a log file. This can be your application or some system daemons like Syslog, Docker log driver or Kubelet, etc.
+2. Tailer - A reader that reads log lines as they are appended e.g agents like Promtail.
+3. Log Rotater - A process that rotates the log file either based on time (say scheduled every day) or size (say a log file reached its max size).
+
+> **NOTE:** Here `fd` defines a file descriptor. Once a file is open for read or write, OS gives back a unique file descriptor (usually an integer) per process, and all the operations like read and write are done over that file descriptor. In other words, once the file is opened successfully, the file descriptor matters more than the file name.
```

Comment:
"OS gives back"  This confused me, do you mean "the operating system gives back a unique file descriptor"?  

Also not sure how to give feedback on a graphic in GitHub, but in logrotation-components.png should there also be a `fd` notation on the line between the Log Rotator and the error.log?

Reply:
> "OS gives back" This confused me, do you mean "the operating system gives back a unique file descriptor"?

Yes.

> Also not sure how to give feedback on a graphic in GitHub, but in logrotation-components.png should there also be a fd notation on the line between the Log Rotator and the error.log?

Technically yes Log Rotator also deals with file descriptor to error.log. But we don't care about how file descriptors are handled in log rotator. That's why left it out.

Reply:
Then I think I would rewrite "the OS gives back" as either 
"The operating system returns a unique file description"   or
"The operating system assigns a unique file description"

Reply:
Went with "The Operating System returns a unique file description"